component/loki/process/stages/drop.go

```diff
@@ -17,22 +17,26 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-// Configuration errors.
-var (
-	ErrDropStageEmptyConfig   = errors.New("drop stage config must contain at least one of `source`, `expression`, `older_than` or `longer_than`")
-	ErrDropStageInvalidConfig = errors.New("drop stage config error, `value` and `expression` cannot both be defined at the same time")
-	ErrDropStageInvalidRegex  = errors.New("drop stage regex compilation error")
+const (
+	ErrDropStageEmptyConfig       = "drop stage config must contain at least one of `source`, `expression`, `older_than` or `longer_than`"
+	ErrDropStageInvalidConfig     = "drop stage config error, `value` and `expression` cannot both be defined at the same time."
+	ErrDropStageInvalidRegex      = "drop stage regex compilation error: %v"
+	ErrDropStageNoSourceWithValue = "drop stage config must contain `source` if `value` is specified"
```

Comment:
I don't really like that these `ErrXYZ` variables are not errors but strings, but it's not a big deal for now.